src/components/CalendarPicker/index.js

```diff
@@ -0,0 +1,169 @@
+            this.setState(prev => ({...prev, currentDateView: moment(prev.currentDateView).set('year', this.props.defaultYear).toDate()}));
+        }
+    }
+
+    /**
+     * Updates the currentDateView state by subtracting one month from it.
+     * @returns {void}
+     */
+    onPrevMonthPressed() {
+        this.setState(prev => ({currentDateView: moment(prev.currentDateView).subtract(1, 'M').toDate()}));
+    }
+
+    /**
+     * Updates the currentDateView state by adding one month to it.
+     * @returns {void}
+     */
+    onNextMonthPressed() {
+        this.setState(prev => ({currentDateView: moment(prev.currentDateView).add(1, 'M').toDate()}));
+    }
+
+    /**
+     * Handles the user pressing the year picker button.
+     * Opens the year selection screen with the minimum and maximum year range
+     * based on the props, the current year based on the state, and the active route.
+     * @returns {void}
+     */
+    onYearPickerPressed() {
+        const minYear = moment(this.props.minDate).year();
+        const maxYear = moment(this.props.maxDate).year();
+        const currentYear = this.state.currentDateView.getFullYear();
+        Navigation.navigate(ROUTES.getYearSelectionRoute(minYear, maxYear, currentYear, Navigation.getActiveRoute()));
+        this.props.onYearPressed();
+    }
+
+    /**
+     * Calls the onSelected function with the selected date, if it is within the min/max range.
+     * @param {number} day - The day of the month that was selected.
+     * @returns {void}
+     */
+    onDayPressed(day) {
+        const selectedDate = new Date(this.state.currentDateView.getFullYear(), this.state.currentDateView.getMonth(), day);
+        this.props.onSelected(selectedDate);
+    }
+
+    render() {
+        const currentMonthView = this.state.currentDateView.getMonth();
+        const currentYearView = this.state.currentDateView.getFullYear();
+        const calendarDaysMatrix = generateMonthMatrix(currentYearView, currentMonthView);
+        const hasAvailableDatesNextMonth = moment(this.props.maxDate).endOf('month').startOf('day') > moment(this.state.currentDateView).add(1, 'M');
+        const hasAvailableDatesPrevMonth = moment(this.props.minDate).startOf('day') < moment(this.state.currentDateView).subtract(1, 'M').endOf('month');
+
+        return (
+            <View>
+                <View style={[styles.calendarHeader, styles.flexRow, styles.justifyContentBetween, styles.alignItemsCenter, styles.ph4, styles.pr1]}>
+                    <TouchableOpacity
+                        onPress={this.onYearPickerPressed}
+                        style={[styles.alignItemsCenter, styles.flexRow, styles.flex1, styles.justifyContentStart]}
+                    >
+                        <Text style={styles.sidebarLinkTextBold} testID="currentYearText" accessibilityLabel={this.props.translate('common.currentYear')}>{currentYearView}</Text>
```

Comment:
Why do we need to pass `testID`?

Reply:
`currentYearText` is used for CalendarPicker component tests in `CalendarPickerTest.js` to check behaviour if the calendar opened with the proper initial year